homeassistant/components/heos/media_player.py

```diff
@@ -1,152 +1,262 @@
+        return None
 
     @property
-    def media_title(self):
-        """Album name of current playing media."""
-        return self._player.media_title
+    def media_position(self):
+        """Position of current playing media in seconds."""
+        # Some media doesn't have duration but reports position, return None
+        if not self._player.now_playing_media.duration:
+            return None
+        return self._player.now_playing_media.current_position / 1000
 
     @property
-    def media_album_name(self):
-        """Album name of current playing media."""
-        return self._player.media_album
+    def media_position_updated_at(self):
+        """When was the position of the current playing media valid."""
+        # Some media doesn't have duration but reports position, return None
+        if not self._player.now_playing_media.duration:
+            return None
+        return self._media_position_updated_at
 
     @property
-    def media_image_url(self):
-        """Return the image url of current playing media."""
-        return self._player.media_image_url
+    def media_image_url(self) -> str:
+        """Image url of current playing media."""
+        return self._player.now_playing_media.image_url
 
     @property
-    def media_content_id(self):
-        """Return the content ID of current playing media."""
-        return self._player.media_id
+    def media_title(self) -> str:
+        """Title of current playing media."""
+        return self._player.now_playing_media.song
 
     @property
-    def is_volume_muted(self):
-        """Boolean if volume is currently muted."""
-        return self._player.mute == "on"
-
-    async def async_mute_volume(self, mute):
-        """Mute volume."""
-        self._player.set_mute(mute)
-
-    async def async_media_next_track(self):
-        """Go TO next track."""
-        self._player.play_next()
+    def name(self) -> str:
+        """Return the name of the device."""
+        return self._player.name
 
-    async def async_media_previous_track(self):
-        """Go TO previous track."""
-        self._player.play_previous()
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed for this device."""
+        return False
 
     @property
-    def supported_features(self):
-        """Flag of media commands that are supported."""
-        return SUPPORT_HEOS
+    def shuffle(self) -> bool:
+        """Boolean if shuffle is enabled."""
+        return self._player.shuffle
 
-    async def async_set_volume_level(self, volume):
-        """Set volume level, range 0..1."""
-        self._player.set_volume(volume * 100)
+    @property
+    def state(self) -> str:
+        """State of the player."""
+        from pyheos import const
+        play_state_to_state = {
+            const.PLAY_STATE_PLAY: STATE_PLAYING,
+            const.PLAY_STATE_STOP: STATE_IDLE,
+            const.PLAY_STATE_PAUSE: STATE_PAUSED
+        }
```

Comment:
@MartinHjelmare Any creative ideas on how to leverage the library constants without having to define the map locally?

Reply:
No, I think it's fine.

Reply:
Good idea.